RESTEASY-2219 Use MP-Context-Propagation instead of Reactive Contexts

FroMage opened the pull request:
~DO NOT MERGE: this is a PR meant for running tests on the CI~

This can be merged now. This switches to SmallRye-CP to handle context propagation, which is the successor of reactive-contexts.

rsvoboda commented on resteasy-dependencies-bom/pom.xml:
```diff
@@ -86,7 +86,7 @@
         <version.weld>3.0.4.Final</version.weld>
         <version.json.patch>1.3</version.json.patch>
-        <version.io.reactiverse.reactive-contexts>1.0.0</version.io.reactiverse.reactive-contexts>
+        <version.io.smallrye.concurrency>1.0.0-SNAPSHOT</version.io.smallrye.concurrency>
```
When do you expect smallrye.concurrency will go 1.0.0(.Final) ?

FroMage replied:
I'm not sure, but it could be within a matter of a few months.

FroMage replied:
Just released 1.0.1

FroMage commented:
Rebased this on master, using MP-CP 1.0-RC1 and latest master of SmallRye-CP. Still don't merge, please, because this is not final. I want to see if CI passes. Tests do pass locally.

Note that I renamed one module `resteasy-reactive-context` to `resteasy-context-propagation` due to the MP spec rename.

FroMage commented:
Well, at least it passes CI.

FroMage commented:
OK, I've updated the code with the latest versions. I think we should merge this ASAP so we can use it in Quarkus. @asoldano ?

FroMage commented:
@asoldano I think you can remove the `do not merge` and `needs rebase` labels too :)

FroMage commented:
Thanks a lot!

asoldano commented:
@FroMage , I'm merging this to the 4.0 branch later today. Then I'll ask you if you can kindly pre-test 4.0.0.Final-SNAPSHOT in Quarkus

FroMage commented:
snapshots built locally, or pushed to a snapshots repo?

asoldano commented:
@FroMage branch ready: https://github.com/resteasy/Resteasy/commits/4.0 
I'd say please build it locally there, the snapshot would go to our nexus repo and I think I remember Quarkus consumes only maven central repo

FroMage commented:
OK lemme build it locally.

FroMage commented:
My test passes so it seems to work.

asoldano commented:
Great, thanks!
